Common/Orders/Fills/FillModel.cs

```diff
@@ -286,6 +303,121 @@ public virtual OrderEvent StopLimitFill(Security asset, StopLimitOrder order)
+        ///     was made the limit fill will be done with closing price of the bar after the trigger has been reached.
+        /// </remarks>
+        public virtual OrderEvent LimitIfTouchedFill(Security asset, LimitIfTouchedOrder order)
+        {
+            //Default order event to return.
+            var utcTime = asset.LocalTime.ConvertToUtc(asset.Exchange.TimeZone);
+            var fill = new OrderEvent(order, utcTime, OrderFee.Zero);
+
+            //If its cancelled don't need anymore checks:
+            if (order.Status == OrderStatus.Canceled) return fill;
+
+            // Fill only if open or extended
+            if (!IsExchangeOpen(asset,
+                Parameters.ConfigProvider
+                    .GetSubscriptionDataConfigs(asset.Symbol)
+                    .IsExtendedMarketHours()))
+            {
+                return fill;
+            }
+            
+            // Get the range of prices in the last bar:
+            var tradeHigh = 0m;
+            var tradeLow = 0m;
+            var bidOpen = 0m;
+            var askOpen = 0m;
+            var pricesEndTime = DateTime.MinValue;
+            
+            var subscribedTypes = GetSubscribedTypes(asset);
+
+            if (subscribedTypes.Contains(typeof(Tick)))
+            {
+                var trade = GetPricesCheckingPythonWrapper(asset, order.Direction);
+                var quote = asset.Cache.GetAll<Tick>().LastOrDefault(x => x.TickType == TickType.Quote && x.Price > 0);
+                
+                if (trade != null)
+                {
+                    tradeHigh = trade.Current;
+                    tradeLow = trade.Current;
+                    bidOpen = quote.BidPrice;
+                    askOpen = quote.AskPrice;
+                    pricesEndTime = trade.EndTime.ConvertToUtc(asset.Exchange.TimeZone);
+                }
+            }
+            
+            else if (subscribedTypes.Contains(typeof(TradeBar)))
+            {
+                var tradeBar = asset.Cache.GetData<TradeBar>();
+                var quoteBar = asset.Cache.GetData<QuoteBar>();
+
+                if (tradeBar != null && quoteBar != null)
+                {
+                    tradeHigh = tradeBar.High;
+                    tradeLow = tradeBar.Low;
+                    bidOpen = quoteBar.Bid.Open;
+                    askOpen = quoteBar.Ask.Open;
+                    pricesEndTime = tradeBar.EndTime.ConvertToUtc(asset.Exchange.TimeZone);
+                }
+            }
+
+            // do not fill on stale data
+            if (pricesEndTime <= order.Time) return fill;
+
+            switch (order.Direction)
+            {
+                case OrderDirection.Sell:
+                    if (tradeHigh >= order.TriggerPrice || order.TriggerTouched)
+                    {
+                        order.TriggerTouched = true;
+
+                        //-> 1.1 Limit surpassed: Sell.
+                        if (tradeHigh >= order.LimitPrice)
```

Comment:
`tradeHigh >= order.LimitPrice` This should be using quotes? Please add a failing unit test for this issue